tests/integration/components/args-decorator-test.ts

```diff
@@ -21,6 +21,18 @@ module('Integration | Component | arg-decorator', function(hooks) {
     assert.dom('.level').hasNoText('@arg will pass `undefined` as a default when an initializer is not present');
   });
 
+  test('it uses the correct context in property initializers', async function(assert) {
+    await render(hbs`
+      <Character @name="link" />
+      <Character @name="zelda" />
+    `);
+
+    const linkId = this.element.querySelector('.character:first-child .id');
```

Comment:
Using splatts would result in less magic-y selectors, but I did not want to modify the current component logic.

Reply:
splatts are good, go for it 👍